client/src/Components/Filter/Filter.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, ChangeEvent, FormEvent } from 'react'
+import { useNavigate } from 'react-router-dom'
+
+import './Filter.css'
+
+const Filter = () => {
+    const [formState, setFormState] = useState({
+        maxPrice: '',
+        includes: '',
+        limit: ''
+    })
+    const navigate = useNavigate()
+
+    const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+        const { name, value } = e.target;
+        setFormState(prev => ({ ...prev, [name]: value }))
+    };
+
+    const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        const queryParams = new URLSearchParams(formState).toString();
```

Comment:
This is just having a little issue with 'maxPrice' vs 'max-price'. We want to use URLSearchParams with max-price, because that's how our back end is set up, and we don't use camelCase in URLs. You can rename the max-price in this call like this: 
```
    const queryParams = new URLSearchParams({ ...formState, "max-price": formState.maxPrice }).toString();
```